test/base/test_store.py

```diff
@@ -141,3 +145,83 @@ def test_storage_trace():
                                 getattr(mem_res['history'],
                                         f'get_{entry}_trace')()[iteration],
                                 hdf5_entry_trace[iteration])
+
+
+def test_storage_profiling():
+    objective = pypesto.Objective(fun=so.rosen,
+                                  grad=so.rosen_der,
+                                  hess=so.rosen_hess)
+    dim_full = 10
+    lb = -5 * np.ones((dim_full, 1))
+    ub = 5 * np.ones((dim_full, 1))
+    n_starts = 5
+    startpoints = pypesto.startpoint.latin_hypercube(n_starts=n_starts,
+                                                     lb=lb,
+                                                     ub=ub)
+    problem = pypesto.Problem(objective=objective, lb=lb, ub=ub,
+                              x_guesses=startpoints)
+
+    optimizer = pypesto.optimize.ScipyOptimizer()
+
+    result_optimization = pypesto.optimize.minimize(
+        problem=problem, optimizer=optimizer,
+        n_starts=n_starts)
+    profile_original = profile.parameter_profile(
+        problem=problem, result=result_optimization,
+        profile_index=[0], optimizer=optimizer)
+    with tempfile.TemporaryDirectory(dir=".") as tmpdirname:
+        _, fn = tempfile.mkstemp(".hdf5", dir=f"{tmpdirname}")
```

Comment:
I think this "with" statement does not make sense here -- what should happen is that a temporary folder is created, then files inserted, or just a single temporary file is created, then the file is filled, and in the end the file is deleted again (can be done either using a with context, or a try--finally.